concur/integrations/glfw.py

```diff
@@ -24,14 +24,24 @@ class PatchedGlfwRenderer(GlfwRenderer):
     keyboard layouts.
 
     See https://github.com/ocornut/imgui/issues/2959 for details.
+
+    # Temporary try except fix until we find a better solution, if we don't apply this,
+    # the app will crash if certain special keys are pressed.
     """
     def keyboard_callback(self, window, key, scancode, action, mods):
-        if key < 0x100:
-            # Translate characters to the correct keyboard layout.
-            key_name = glfw.get_key_name(key, 0)
-            if key_name is not None:
-                key = ord(key_name.upper())
-        super(PatchedGlfwRenderer, self).keyboard_callback(window, key, scancode, action, mods)
+        try:
+            _key = key
+            if _key < 0x100:
+                # Translate characters to the correct keyboard layout.
+                key_name = glfw.get_key_name(key, 0)
+                if key_name is not None:
+                    _key = ord(key_name.upper())
+            super(PatchedGlfwRenderer, self).keyboard_callback(window, _key, scancode, action, mods)
+        except:
+            try:
+                super(PatchedGlfwRenderer, self).keyboard_callback(window, key, scancode, action, mods)
+            except:
```

Comment:
Is this second `try` necessary? When does this line throw an exception?

Reply:
Basically no, I put this one in a try as well to be very sure the app will  not crash just because someone got a very special keyboard.

Reply:
OK, could you remove this second `try`? I think it's better to crash (and then fix the issue), rather than silently ignore keypresses. The cause may then be hard to find.

The first `try` is needed, really, because `PatchedGlfwRenderer` misuses GLFW. I have an exotic keyboard myself, and it was the only way I found to fix my keyboard shortcuts. The normal keyboard callback should be fine.

Reply:
Agreed, I will remove the second try